sql/core/src/main/scala/org/apache/spark/sql/sources/PartitioningUtils.scala

```diff
@@ -35,6 +35,10 @@ private[sql] case class Partition(values: Row, path: String)
 private[sql] case class PartitionSpec(partitionColumns: StructType, partitions: Seq[Partition])
 
 private[sql] object PartitioningUtils {
+  // This duplicates default value of Hive `ConfVars.DEFAULTPARTITIONNAME`, since sql/core doesn't
+  // depend on Hive.
+  private[sql] val DEFAULT_PARTITION_NAME = "__HIVE_DEFAULT_PARTITION__"
```

Comment:
Not sure whether this is a good idea, since we are trying to be Hive compatible... Definitions from official Hive code should be the "standard" here.